src/coverlet.core/Helpers/InstrumentationHelper.cs

```diff
@@ -121,12 +121,13 @@ public bool EmbeddedPortablePdbHasLocalSource(string module, out string firstNot
                             {
                                 Document document = metadataReader.GetDocument(docHandle);
                                 string docName = _sourceRootTranslator.ResolveFilePath(metadataReader.GetString(document.Name));
-
+                                Guid languageGuid = metadataReader.GetGuid(document.Language);
                                 // We verify all docs and return false if not all are present in local
                                 // We could have false negative if doc is not a source
                                 // Btw check for all possible extension could be weak approach
-                                // We exlude from the check the autogenerated source file(i.e. source generators)
-                                if (!_fileSystem.Exists(docName) && !docName.EndsWith(".g.cs"))
+                                // We exclude from the check the autogenerated source file(i.e. source generators)
+                                // We exclude special F# construct https://github.com/coverlet-coverage/coverlet/issues/1145
+                                if (!_fileSystem.Exists(docName) && !docName.EndsWith(".g.cs") && !IsUnknownModuleInFSharpAssembly(languageGuid, docName))
```

Comment:
This logic could be extracted into a method, since it is identical with the logic in `PortablePdbHasLocalSource`. This helps if we need to add even more exceptions, e.g. for the null documents in VB (#1073).

Reply:
Sure, done.